build-support/bin/external_tool/python.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import os
+import json
+import textwrap
+from typing import TypeVar, Any, Generator
+from pathlib import Path
+import ast
+
+logger = logging.getLogger(__name__)
+T = TypeVar("T")
+
+
+def get_class_variables(file_path: Path, class_name: str, *, variables: type[T]) -> T:
+    """Reads a Python file and retrieves the values of specified class variables."""
+
+    logger.info("parsing %s variables in %s", class_name, file_path)
+    with open(file_path, "r", encoding="utf-8") as file:
+        source_code = file.read()
+
+    tree = ast.parse(source_code)
```

Comment:
we repeat the opening and parsing of files, and these functions also have similar logic to find or replace class vars. Is it worth finding the commonality and extracting? Is there a specific reason for doing it this way, or was this just quicker to get working?

Reply:
No, it was this just quicker to get working. The longest thing is fetching files from github anyway, so I didn't try to optimize this